docs/source/api.rst

```diff
@@ -151,7 +151,165 @@ Closing a driver will immediately shut down all connections in the pool.
     query, use :meth:`neo4j.Driver.verify_connectivity`.
 
 .. autoclass:: neo4j.Driver()
-   :members: session, encrypted, close, verify_connectivity, get_server_info
+    :members: session, query_bookmark_manager, encrypted, close,
+              verify_connectivity, get_server_info
+
+    .. method:: execute_query(query, parameters=None,routing=neo4j.RoutingControl.WRITERS, database=None, impersonated_user=None, bookmark_manager=self.query_bookmark_manager, result_transformer=Result.to_eager_result, **kwargs)
```

Comment:
There are plenty of errors this can raise. Here are the one's I can think of on the spot
 * `Neo4jError` (or any sub-class) - anything the server complains about
 * `SessionExpired`, `ServiceUnavailable` (or any sub-class) - lost/couldn't establish connectivity
 * `ConfigurationError` `TypeError` `ValueError` (maybe even `IndexError`?) - invalid configuration/parameters or trying to use a feature that's too new for the server you're connected to
 * `BoltError` (or any sub-class; internal error class you shouldn't ever encounter) - the server violated the protocol
 
 There might be more that I can't think of right now. My conclusion:
 * All of there errors are fatal (there is pretty much no benefit in giving them different treatment). The user might as well just catch any exception, maybe log it or whatever they feel doing, and then give up (that could mean crash, surface some error to the end-user, whatever is suitable). Plus maybe fixing their code afterwards.
 * I can hardly give any guarantees what errors this method can raise as it pretty much exercises the whole stack of the driver. It's highly likely that I'll forget some corner of the driver and if not, that this piece of documentation will unsync sooner or later as it's so far up the abstraction stack.
 
 So much for why I didn't include it. If you think it'd be helpful nonetheless, we can make it happen.

Reply:
> The user might as well just catch any exception

:+1: